### PR TITLE
Testing improvement: Create snapshots for geolocation errors

### PR DESCRIPTION
**Pull request by dieghernan**
Following comment on #535  by @hadley , any guidance would be appreciated.

Regards

### CODE REVIEW

File: tests/testthat/test-post-tweet.R [hunk 1]
```diff
@@ -53,3 +45,12 @@ test_that("post_tweet geolocated works", {
   rt <- expect_message(post_destroy(crt$id_str), "your tweet has been deleted!")
   expect_equal(httr::status_code(rt), 200L)
 })
+
+test_that("post_tweet geolocated validity checks", {
+  msg <- paste("test geolocated error", Sys.time()) # To avoid having duplicated status
+  expect_snapshot(post_tweet(msg, lat = "x", long = 0), error = TRUE)
```
Comment (hadley) [inline tests/testthat/test-post-tweet.R:51]: For this type of snapshotting, I'd put them all in one snapshot, like:

```R
expect_snapshot(error = TRUE, {
  ...
})
```

File: tests/testthat/_snaps/post-tweet.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      post_tweet(msg, lat = 91, long = 0)
+    Error <simpleError>
+      lat should be between -90 and 90 degrees
+
+---
+
+    Code
+      post_tweet(msg, lat = 0, long = 181)
+    Error <simpleError>
+      long should be between -180 and 180  degrees
```
Comment (hadley) [inline tests/testthat/_snaps/post-tweet.md:27]: extra space before degrees

File: tests/testthat/_snaps/post-tweet.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# post_tweet geolocated validity checks
+
+    Code
+      post_tweet(msg, lat = "x", long = 0)
+    Error <simpleError>
+      lat should be numeric
```
Comment (hadley) [inline tests/testthat/_snaps/post-tweet.md:6]: Should put `` ` `` around argument names

File: tests/testthat/test-post-tweet.R (hunk 1 above)
Comment (dieghernan) [reply tests/testthat/test-post-tweet.R]: Changed to the proposed approach

**Comment by dieghernan** [comment]
Hi @hadley , I think I have tackled everything you mentioned, regards

File: R/post-tweet.R [hunk 4]
```diff
@@ -131,15 +131,15 @@ post_tweet <- function(status = "my first rtweet #rstats",
   ## geotag if provided
   if (!is.null(lat) && !is.null(long)) {
     # Validate inputs
-    if (!is.numeric(lat)) stop("lat should be numeric")
-    if (!is.numeric(long)) stop("long should be numeric")
+    if (!is.numeric(lat)) stop("`lat` should be numeric.")
```
Comment (hadley) [inline R/post-tweet.R:134]: Sorry, just realised that these would be better as "must", not "should"

Comment (dieghernan) [reply R/post-tweet.R]: No worries, done

**Comment by dieghernan** [comment]
@hadley I think it is ready now

**Pull request merged**